### DISCUSSION TITLE
Session scope

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [reagento/dishka#241] [Q&A] [answered]
`RUNTIME] → APP → [SESSION] → REQUEST → ACTION → STEP`

A user makes multiple requests during their session. The previous request is stored as a dependency injection ie. if request#2 is sent, request#1 is read from the DI. On completion, request#1 is overwritten with request#2 in the DI and so on until the user exits the session. Is this possible with dishka?

**Comment by Tishka17** [comment]
It is not clear how are you going to use those requests. Another question is whether requests can be concurrent or not

**Reply by dineshbvadhia** [reply]
Eg. A user makes multiple requests to the server during their session. The function processing request#(n+1) will use information extracted from request#(n) (stored as a json) ie. the previous request is not executed by the current request. 

The DI stores the current request data as json so it is available to the next request. The requests are sequential. Does this help?

**Reply by Tishka17** [reply]
I guess the library does not provide solution for that case. Probably, you can create some session-scoped object and store request there manually.

**Reply by dineshbvadhia** [reply]
I'm new to dishka so, how would I start on this or is there a base example to build on? Thanks

**Reply by Tishka17** [reply]
What framework are you using? What exactly is session in youd case?

**Reply by dineshbvadhia** [reply]
FastAPI - latest version.

Fastapi Request is used for path operations.

The software runs on fastapi. A user request (payload) is sent from the client to the server which fastapi routes. There will be n x users. A custom session is not being created.

**Reply by Tishka17** [reply, marked as answer]
Ok, it sounds a bit away from problems which are solved by dishka. 

In dishka Scope.SESSION is something which exists in your app while other requests are running. It is ok for websocket connections or something like that. 
In regular web apps HTTP-Session is **data** which is shared among requests. You can have multiple application instances, they cvan be restarted. Session exists regardless on that. 

So, I suppose you have to deal with this date in your application or presentation logic
